tests/test_study.py

```diff
@@ -425,18 +425,23 @@ def func_nan(_):
 # TODO(Yanase): Remove this test function after removing `optuna.structs.TrialPruned`.
 @pytest.mark.parametrize('trial_pruned_class', [optuna.exceptions.TrialPruned,
                                                 optuna.structs.TrialPruned])
-def test_run_trial_with_trial_pruned(trial_pruned_class):
-    # type: (Callable[[], optuna.exceptions.TrialPruned]) -> None
+@pytest.mark.parametrize('report_value', [None, 1.2])
+def test_run_trial_with_trial_pruned(trial_pruned_class, report_value):
+    # type: (Callable[[], optuna.exceptions.TrialPruned], Optional[float]) -> None
 
     study = optuna.create_study()
 
-    def func_with_trial_pruned(_):
+    def func_with_trial_pruned(trial):
         # type: (optuna.trial.Trial) -> float
 
+        if report_value is not None:
+            trial.report(report_value, 1)
+
```

Comment:
It's possible to test that the value of the trial is not set. Open for suggestions. I personally feel that that test would look quite historical and not necessarily comprehensible.

Reply:
>  test that the value of the trial is not set.

IMO, it is reasonable to keep the test case, although the test function can be much simpler without it.
For example, some users use the pruning feature for constrained optimization. In that case, no intermediate value will be reported. Please take a look at the following code. It checks the validity of the suggested parameter set and prunes the trial if it is invalid.

```python
def objective(trial):
    x = trial.suggest_uniform('x', 0, 1)
    if not is_valid(x):
        raise TrialPruned()
    ...
```